Mathlib/NumberTheory/LSeries/Convolution.lean

```diff
@@ -122,28 +117,27 @@ lemma term_convolution' (f g : ℕ → ℂ) (s : ℂ) :
     -- the right hand sum is over the union below, but in each term, one factor is always zero
     have hS : (fun p ↦ p.1 * p.2) ⁻¹' {0} = {0} ×ˢ univ ∪ univ ×ˢ {0} := by
       ext
-      simp only [mem_preimage, mem_singleton_iff, Nat.mul_eq_zero, mem_union, mem_prod, mem_univ,
-        and_true, true_and]
+      simp [Nat.mul_eq_zero,  -singleton_prod, -prod_singleton]
     have : ∀ p : (fun p : ℕ × ℕ ↦ p.1 * p.2) ⁻¹' {0}, term f s p.val.1 * term g s p.val.2 = 0 := by
-      rintro ⟨⟨p₁, p₂⟩, hp⟩
-      rcases hS ▸ hp with ⟨rfl, -⟩ | ⟨-, rfl⟩ <;> simp only [term_zero, zero_mul, mul_zero]
-    simp only [this, tsum_zero]
+      rintro ⟨⟨_, _⟩, hp⟩
+      rcases hS ▸ hp with ⟨rfl, -⟩ | ⟨-, rfl⟩ <;> simp
+    simp [this]
   -- now `n ≠ 0`
   rw [show (fun p : ℕ × ℕ ↦ p.1 * p.2) ⁻¹' {n} = n.divisorsAntidiagonal by ext; simp [hn],
     Finset.tsum_subtype' n.divisorsAntidiagonal fun p ↦ term f s p.1 * term g s p.2,
     term_convolution f g s n]
 
 end LSeries
 
+-- NB: terminal `simpa` is quite slow if un-squeezed
 open Set in
 /-- The L-series of the convolution product `f ⍟ g` of two sequences `f` and `g`
 equals the product of their L-series, assuming both L-series converge. -/
 lemma LSeriesHasSum.convolution {f g : ℕ → ℂ} {s a b : ℂ} (hf : LSeriesHasSum f s a)
     (hg : LSeriesHasSum g s b) :
     LSeriesHasSum (f ⍟ g) s (a * b) := by
-  simp only [LSeriesHasSum, term_convolution']
   have hsum := summable_mul_of_summable_norm hf.summable.norm hg.summable.norm
-  exact (HasSum.mul hf hg hsum).tsum_fiberwise (fun p ↦ p.1 * p.2)
+  simpa only [LSeriesHasSum, term_convolution'] using (hf.mul hg hsum).tsum_fiberwise _
```

Comment:
Could we move the note to immediately before the `simpa`?

Reply:
Sure, done